airflow/models/taskinstance.py

```diff
@@ -886,6 +889,7 @@ def refresh_from_task(self, task: Operator, pool_override: str | None = None) ->
         # value that needs to be stored in the db.
         self.executor_config = task.executor_config
         self.operator = task.task_type
+        self.custom_operator_name = task.operator_name
```

Comment:
I think the same thing is accomplished right now as well, since the task.operator_name handling the case of Null value at the operator class end.

 But think the current approach is confusing. I'll make the change.

Reply:
Pushed the changes, please check

Reply:
In the current approach, _old_ rows in `task_instance` (that existed before the migration) would have NULL and you still need to handle those anyway. So it’s a bit inconsistent.

Reply:
@uranusjr As per your suggestion now I have added a property `operator_name` in the task_instance model, which will take care of the NULL case. 

Now, we are populating ti's `cutom_operator_name` with operator's `cutom_operator_name`. Then in `operator_name` we are falling back to ti's operator if `cutom_operator_name` is NULL.